arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/continuations/Builders.kt

```diff
@@ -87,18 +95,9 @@ public class IorShift<E> @PublishedApi internal constructor(semigroup: Semigroup
         rightValue
       }
     }
-  
-  @PublishedApi
-  internal inline operator fun <A> invoke(action: IorShift<E>.() -> A): Ior<E, A> {
-    val res = action(this)
-    val leftState = leftState.get()
-    return if (leftState === EmptyValue) Ior.Right(res)
-    else Ior.Both(EmptyValue.unbox(leftState), res)
-  }
-  
-  @Suppress("UNCHECKED_CAST")
+
   private fun combine(other: E): E =
-    leftState.updateAndGet { state ->
-      if (state === EmptyValue) other else EmptyValue.unbox<E>(state).combine(other)
```

Comment:
do we still need `EmptyValue` after these changes?

Reply:
I'd need to check, maybe it's not used anywhere anymore after this change. I choose `Option` here, but the `Option` allocations could be removed using `EmptyValue` again.